tools/src/faircv/Main.hs

```diff
@@ -1,26 +1,31 @@
-
 import Data.Aeson
-import qualified Data.Set as Set
-import Data.Time.Clock
-import Data.Traversable
 import Servant.Client
 import Test.QuickCheck
+import Time.Units (threadDelay, sec)
 
 import Loot.Log
 
 import Dscp.Core
 import Dscp.Crypto
+import Dscp.Educator.Web.Student
+import Dscp.Educator.Web.Types
 import Dscp.Resource.Keys
+import Dscp.Util (leftToFail, nothingToFail)
+import Dscp.Util.Test (detGenG)
 import Dscp.Witness.Web hiding (checkFairCV)
 
 import Client
-
-createActor :: IO (SecretKey, PublicKey, Address)
-createActor = do
-    sk   <- genSecretKey
-    pk   <- return (toPublic sk)
-    addr <- return (mkAddr   pk)
-    return (sk, pk, addr)
+import Options
+
+-- This is a copy of dscp-keygen's parseInputWithSecret (for SecretFromSeed)
+secretFromSeed :: ByteString -> Maybe SecretKey
+secretFromSeed input = asum [
+      do
+        sd <- readMaybe @Word . toString @Text $ decodeUtf8 input
+        return $ withIntSeed (fromIntegral sd) genSecretKey
+    , do
+        return $ withSeed input genSecretKey
+    ]
```

Comment:
Would be very nice to move this (and the original function) somewhere to `Dscp.Crypto.?` with a note "this is how we generate secret keys from user-supplied seed".
I did this once, but then my change occurred to be redundant.

Reply:
Done, I moved the function in `Dscp.Crypto.Util` which seemed a fair place.

Reply:
Thanks :+1: